diff -up smartmontools-5.38/scsiprint.cpp.perc smartmontools-5.38/scsiprint.cpp
--- smartmontools-5.38/scsiprint.cpp.perc	2008-03-04 23:09:47.000000000 +0100
+++ smartmontools-5.38/scsiprint.cpp	2008-03-27 11:28:20.000000000 +0100
@@ -1037,6 +1037,9 @@ static int scsiGetDriveInfo(int device, 
 	       (0 == strncmp((const char *)(gBuf + 36), "MVSATA", 6))) {
         pout("please try '-d marvell'\n");
         return 2;
+    } else if  (0 == strncmp(product, "PERC", 4)) {
+        pout("\nDELL PERC controllers are not supported.\n");
+        return 2;
     } else if ((0 == con->controller_explicit) &&
 	       (0 == strncmp(manufacturer, "ATA     ", 8)) &&
                has_sat_pass_through(device, 0)) {
