--- smartmontools-5.37/smartctl.cpp.3ware	2006-10-09 13:45:12.000000000 +0200
+++ smartmontools-5.37/smartctl.cpp	2007-08-20 13:20:24.000000000 +0200
@@ -463,8 +463,8 @@
             if (split_report_arg2(s, &i)) {
                  sprintf(extraerror, "Option -d 3ware,N requires N to be a non-negative integer\n");
                  badarg = TRUE;
-            } else if (i<0 || i>15) {
-                 sprintf(extraerror, "Option -d 3ware,N (N=%d) must have 0 <= N <= 15\n", i);
+            } else if (i<0 || i>23) {
+                 sprintf(extraerror, "Option -d 3ware,N (N=%d) must have 0 <= N <= 23\n", i);
                  badarg = TRUE;
             } else {
  	        // NOTE: controller_port == disk number + 1
--- smartmontools-5.37/smartd.cpp.3ware	2006-11-12 05:49:09.000000000 +0100
+++ smartmontools-5.37/smartd.cpp	2007-08-20 13:20:45.000000000 +0200
@@ -3038,8 +3038,8 @@
               PrintOut(LOG_CRIT, "File %s line %d (drive %s): Directive -d 3ware,N requires N integer\n",
                	       configfile, lineno, name);
               badarg=1;
-          } else if ( i<0 || i>15) {
-              PrintOut(LOG_CRIT, "File %s line %d (drive %s): Directive -d 3ware,N (N=%d) must have 0 <= N <= 15\n",
+          } else if ( i<0 || i>23) {
+              PrintOut(LOG_CRIT, "File %s line %d (drive %s): Directive -d 3ware,N (N=%d) must have 0 <= N <= 23\n",
                        configfile, lineno, name, i);
               badarg=1;
           } else {
--- smartmontools-5.37/smartd.conf.5.in.3ware	2006-12-20 08:30:43.000000000 +0100
+++ smartmontools-5.37/smartd.conf.5.in	2007-08-20 13:16:39.000000000 +0200
@@ -298,10 +298,10 @@
 
 .I 3ware,N
 \- the device consists of one or more ATA disks connected to a 3ware
-RAID controller. The non-negative integer N (in the range from 0 to 15
+RAID controller. The non-negative integer N (in the range from 0 to 23
 inclusive) denotes which disk on the controller is monitored.  In log
 files and email messages this disk will be identified as 3ware_disk_XX
-with XX in the range from 00 to 15 inclusive.
+with XX in the range from 00 to 23 inclusive.
 
 This Directive may at first appear confusing, because the 3ware
 controller is a SCSI device (such as /dev/sda) and should be listed as
@@ -729,7 +729,7 @@
 is set to the device path (examples: /dev/hda, /dev/sdb).
 .IP \fBSMARTD_DEVICETYPE\fP 4
 is set to the device type (possible values: ata, scsi, 3ware,N, hpt,L/M/N).
-Here N=0,...,15 denotes the ATA disk behind a 3ware RAID controller and
+Here N=0,...,23 denotes the ATA disk behind a 3ware RAID controller and
 L/M/N denotes the SATA disk behind a HighPoint RocketRAID controller.
 .IP \fBSMARTD_DEVICESTRING\fP 4
 is set to the device description.  For SMARTD_DEVICETYPE of ata or
